### ISSUE TITLE
 Warn about a potentially misplaced exclamation sign

### ISSUE THREAD

**Issue by laino** [eslint/eslint#11801]
**Please describe what the rule should do:**

Warn about a potentially misplaced exclamation sign.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
if (someThing.withAn.arrayProperty.length ==! 1) { // Probably an error!
}
````

```js
if (someValue ==! 1) { // Probably an error!
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

This seems like a basic thing.

**Are you willing to submit a pull request to implement this rule?**

Maybe on a weekend.

**Comment by g-plane** [comment]
Hey @laino , thanks for the issue.

I think enabling the [eqeqeq](https://eslint.org/docs/rules/eqeqeq) rule can solve your problem.

**Comment by laino** [comment]
@g-plane  Both ```space-infix-ops``` and ```space-unary-ops``` should catch this already.

My eslint config was broken though. Closing this.
